apps/aecontract/test/aecontract_SUITE.erl

```diff
@@ -4579,6 +4582,52 @@ sophia_crypto(_Cfg) ->
     ?assertMatchFATE({bytes, Sha256},    ResSha256),
     ?assertMatchFATE({bytes, Blake2b},   ResBlake2b),
 
+    %% Tests for bytes (which should hash as raw strings from lima)
+
+    case vm_version() >= ?VM_AEVM_SOPHIA_3 of   %% No bytes in minerva
+        false -> ok;
+        true  ->
+            Bytes1 = {bytes, <<"ARRAY OF 17 BYTES">>},
+            Bytes2 = {bytes, <<"A LONGER ARRAY OF 52 BYTES (CRUCIALLY, MORE THAN 32)">>},
+
+            BytesToHash = fun({bytes, Bytes}) ->
+                case vm_version() of
+                    VM when ?IS_FATE_SOPHIA(VM)     -> Bytes;
+                    VM when VM >= ?VM_AEVM_SOPHIA_4 -> Bytes;
+                    _ -> aeb_heap:to_binary(format_aevm_args({bytes, Bytes}))
+                end end,
+
+            <<Sha3_B1_N:256>>    = Sha3_B1    = aec_hash:hash(evm, BytesToHash(Bytes1)),
+            <<Sha256_B1_N:256>>  = Sha256_B1  = aec_hash:sha256_hash(BytesToHash(Bytes1)),
+            <<Blake2b_B1_N:256>> = Blake2b_B1 = aec_hash:blake2b_256_hash(BytesToHash(Bytes1)),
+
+            <<Sha3_B2_N:256>>    = Sha3_B2    = aec_hash:hash(evm, BytesToHash(Bytes2)),
+            <<Sha256_B2_N:256>>  = Sha256_B2  = aec_hash:sha256_hash(BytesToHash(Bytes2)),
+            <<Blake2b_B2_N:256>> = Blake2b_B2 = aec_hash:blake2b_256_hash(BytesToHash(Bytes2)),
+
+            ResSha3_B1    = ?call(call_contract, Acc, IdC, sha3_b17,    word, {Bytes1}),
+            ResSha256_B1  = ?call(call_contract, Acc, IdC, sha256_b17,  word, {Bytes1}),
+            ResBlake2b_B1 = ?call(call_contract, Acc, IdC, blake2b_b17, word, {Bytes1}),
+
+            ResSha3_B2    = ?call(call_contract, Acc, IdC, sha3_b52,    word, {Bytes2}),
+            ResSha256_B2  = ?call(call_contract, Acc, IdC, sha256_b52,  word, {Bytes2}),
+            ResBlake2b_B2 = ?call(call_contract, Acc, IdC, blake2b_b52, word, {Bytes2}),
+
+            ?assertMatchAEVM1OOG(Sha3_B1_N   , ResSha3_B1),
```

Comment:
could be just one sequence of `?assertMatchVM`?

Reply:
Yeah I guess. I copied the existing pattern.